### PR TITLE
ignore nan values when transforming 3d coords into colours

### LINKED ISSUE #6870: plot_psd with spatial_colors=True fails when some location values are nan
Adding a simple warning would help, it took me a while to realize that this was the reason my plot was all in black. Even better would be plotting only the nan electrodes in black and showind them beside the head in the little head color plot.

### PR DESCRIPTION
**Pull request by dtxe**
…for spatial_colors feature

#### Reference issue
Fixes #6870

#### What does this implement/fix?
Evoked butterfly plots with spatial_colors=True will plot all traces in black when any of the DigMontage coordinates are nan. The expected behaviour is that sensors with nan coordinates will be in black, while the remaining traces will be coloured.

This fix uses `np.nanmin` and `np.nanmax` instead of `np.amin` and `np.amax` when transforming x,y,z coordinates into colours, resulting in the expected behaviour.

### CODE REVIEW

**Comment by agramfort** [comment]
@dtxe thx. Can you update or add a test for this? 🙏

**Comment by dtxe** [comment]
> @dtxe thx. Can you update or add a test for this? 🙏

Will upload this next week!

**Comment by larsoner** [comment]
@dtxe can you add a test in the next few days and update `latest.inc`? Then we can get this into 1.2 which we plan to release next week.

If you can't get to it, I could push a quick commit if you want

**Comment by dtxe** [comment]
Tests and updated latest.inc file are done!

The updated test will manually set the channel locations of one gradiometer to nan, then check that not all the resulting plot traces will have nan or black colours. 

Please let me know if a different testing strategy is preferred, and I'd be happy to update it.

File: doc/changes/latest.inc [hunk 1]
```diff
@@ -61,6 +61,7 @@ Bugs
 - Fix bug in :func:`mne.io.read_raw_bti` where EEG, EMG, and H/VEOG channels were not detected properly, and many non-ECG channels were called ECG. The logic has been improved, and any channels of unknown type are now labeled as ``misc`` (:gh:`11102` by `Eric Larson`_)
 - Fix bug in :func:`mne.viz.plot_topomap` when providing ``sphere="eeglab"`` (:gh:`11081` by `Mathieu Scheltienne`_)
 - The string and HTML representation of :class:`mne.preprocessing.ICA` reported incorrect values for the explained variance. This information has been removed from the representations, and should instead be retrieved via the new :meth:`~mne.preprocessing.ICA.get_explained_variance_ratio` method (:gh:`11141` by `Richard Höchenberger`_)
+- Fix bug in :meth:`mne.Evoked.plot` and related methods where a ``np.nan`` location value in any channel causes spatial colours to fail (:gh:`6870` by `Simeon Wong`_)
```
Comment (larsoner) [inline doc/changes/latest.inc:64]: FYI @dtxe I pushed a commit to only reference publicly documented functions/methods in `latest.inc`. Users don't care or know about `mne.viz._rgb` even though that's what you had to change, so we make it clear that the impact on them is that things like `evoked.plot` instead

**Review by larsoner — APPROVED** [review]
LGTM, marking for merge -- thanks @dtxe !

**Pull request merged**